core/src/lib/internal/net/BasicRequest.ts

```diff
@@ -1,11 +1,30 @@
 import Method from "./Method";
 
-export default class BasicRequest {
-  public static Builder = class {
+export class BasicRequest {
+  public readonly method: Method;
+  public readonly url: string;
+  public readonly headers: { [key: string]: string };
+  public readonly body?: string;
```

Comment:
Per the discussion in Slack earlier, I made request bodies and response bodies strings instead of `any`s so the typescript compiler actually does type checking for us, and it was able to catch a bunch of issues where we weren't decoding or encoding JSON into or from a string in several places.